### PR TITLE
Migrate CRDs to apiextensions.k8s.io/v1

### PR DESCRIPTION
**Pull request by evankanderson**
I didn't fill in the OpenAPI schema, so you won't get any APIServer-provided validation, but this should work for v1.15+ apiservers (at least, I didn't check when the v1 CustomResourceDefinition was introduced).

```release-note
:broom: Migrate to CRD v1 scheme
```

### CODE REVIEW

File: config/300-vspherebinding.yaml [hunk 1]
```diff
@@ -22,12 +21,21 @@ spec:
...
-    JSONPath: ".status.conditions[?(@.type=='Ready')].reason"
+  versions:
+  - name: v1alpha1
+    served: true
+    storage: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        x-kubernetes-preserve-unknown-fields: true
+        # TODO: use controller-gen from controller-tools to fill this in?
```
Comment (embano1) [inline config/300-vspherebinding.yaml:32]: What's the current pattern in Knative projects to create schemas?

**Review by embano1 — APPROVED** [review]
LGTM

File: config/300-vspherebinding.yaml (hunk 1 above)
Comment (n3wscott) [reply config/300-vspherebinding.yaml:32]: we have been using this: https://github.com/knative/eventing/tree/main/cmd/schema

Comment (n3wscott) [reply config/300-vspherebinding.yaml:32]: tracking https://github.com/vmware-tanzu/sources-for-knative/issues/253

**Review by n3wscott — APPROVED** [review]

**Pull request merged**